edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/Avaliability.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Avaliability
```

Comment:
class name Availability is very general, what availability is? module availability?

Reply:
This class is just generic availability. It holds and calculates uptime vs. downtime. It could be anything.

Reply:
In general, I think it is simple to just report whether a component is up or down as metric; and this can be used for graph or calculating overall %, etc.  Please check with Damon which one is better.